daemon/c_vol.c

```diff
@@ -858,9 +848,78 @@ c_vol_cleanup_dm(c_vol_t *vol)
 	return 0;
 }
 
+static int
+c_vol_umount_dir(const char *mount_dir)
+{
+	IF_NULL_RETVAL(mount_dir, -1);
+
+	while (file_is_mountpoint(mount_dir)) {
+		if (umount(mount_dir) < 0) {
+			if (umount2(mount_dir, MNT_DETACH) < 0) {
+				ERROR_ERRNO("Could not umount '%s'", mount_dir);
+				return -1;
+			}
+		}
+	}
+	return 0;
+}
+
+/**
+ * Umount all image files.
+ * This function is called in the rootns, to cleanup stopped container.
+ */
+static int
+c_vol_umount_all(c_vol_t *vol)
+{
+	int i, n;
+	char *root = mem_printf("/tmp/%s", uuid_string(container_get_uuid(vol->container)));
+
+	bool setup_mode = container_get_state(vol->container) == CONTAINER_STATE_SETUP;
```

Comment:
call for stopped container => always _STOPPED ?

Reply:
fixed in latest push